### PR TITLE
feat: [REPUX-722] make current balance auto updatable

### PR DESCRIPTION
**Pull request by davo22**

### CODE REVIEW

File: src/app/marketplace/marketplace-cancel-purchase-button/marketplace-cancel-purchase-button.component.ts [hunk 1]
```diff
@@ -48,11 +54,11 @@ export class MarketplaceCancelPurchaseButtonComponent implements OnInit, OnDestr
   }
 
   get isDisabled(): boolean {
-    if (!this.dataProduct.blockchainState) {
+    if (!this.blockchainDataProduct) {
```
Comment (kduralek) [inline src/app/marketplace/marketplace-cancel-purchase-button/marketplace-cancel-purchase-button.component.ts]: wouldn't be easier to use: `return this.blockchainDataProduct && this.blockchainDataProduct.disabled`?

File: src/app/services/wallet.service.ts [hunk 2]
```diff
@@ -24,6 +25,7 @@ export class WalletService implements OnDestroy {
   private rafReference: number;
   private metamaskStatusSubject = new BehaviorSubject<MetamaskStatus>(undefined);
   private walletSubject = new BehaviorSubject<Wallet>(undefined);
+  private balanceSubject = new BehaviorSubject<BigNumber>(undefined);
```
Comment (kduralek) [inline src/app/services/wallet.service.ts:28]: Do we really need separate subject for `wallet` property? Wouldn't be proper to just use existing `walletSubject`?

File: src/app/shared/components/metamask-detector/metamask-detector.component.html [hunk 3]
```diff
@@ -1,3 +1,3 @@
-<div *ngIf="status && status !== statuses.Ok" class="metamask-detector__banner">
+<div *ngIf="statusIsDefined && status !== statuses.Ok" class="metamask-detector__banner">
```
Comment (kduralek) [inline src/app/shared/components/metamask-detector/metamask-detector.component.html]: `status !== undefined`

**Review by kduralek — APPROVED** [review]

File: src/app/services/wallet.service.ts (hunk 2 above)
Comment (davo22) [reply src/app/services/wallet.service.ts:28]: A lot of components subscribes for wallet address changes. These components make some requests to metaindexer and to blockchain. If I emit event on walletSubject when balance is updated all these components will unnecessary resend all requests.

Comment (kduralek) [reply src/app/services/wallet.service.ts:28]: ok, got it

**Pull request merged**